test/Microsoft.AspNetCore.TestHost.Tests/TestClientTests.cs

```diff
@@ -239,6 +239,54 @@ public void Dispose()
             clientSocket.Dispose();
         }
 
+        [ConditionalFact]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Hangs randomly (issue #507)")]
```

Comment:
I think there are many in that file, we should make a separate issue to remove mono test checks.